src/hibernate-resume/meson.build

```diff
@@ -11,12 +11,11 @@ executables += [
                         'hibernate-resume-generator.c',
                         'hibernate-resume-config.c',
                 ),
+                'extract' : files('hibernate-resume-config.c'),
         },
         libexec_template + {
                 'name' : 'systemd-hibernate-resume',
-                'sources' : files(
-                        'hibernate-resume.c',
-                        'hibernate-resume-config.c',
-                ),
+                'sources' : files('hibernate-resume.c'),
+                'objects' : ['systemd-hibernate-resume-generator'],
```

Comment:
pff, this one feels just obnoxious.

For tests this makes sense, but otherwise let's not go overboard with extraction, because to me it implies some sort of belonging/ownership where there's none.

Reply:
Compiling files twice is just wasteful, if you don't want this, let's move the shared code to libshared so that it's only compiled once.

Alternatively we introduce a static library but that's just so much more boilerplate than just extracting the objects.

Reply:
I really wish build systems had provided implicit caches that can be opt in via a simple knob + same .c file.

But at least can we treat hibernate-resume.c as the main one?